src/pages/InterviewSchedulingPage/components/SchedulersWithConfirmButton.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState, useEffect } from 'react';
+
+import InterviewScheduler from './InterviewScheduler';
+import ConfirmScheduleButton from './ConfirmScheduleButton';
+
+interface SchedulersWithConfirmButtonProps {
+  startDates: Date[];
+}
+
+function SchedulersWithConfirmButton({
+  startDates,
+}: SchedulersWithConfirmButtonProps): JSX.Element {
+  const [dirty, setDirty] = useState(false);
+  const [userSchedules, setUserSchedules] = useState<Date[][]>([[]]);
+
+  useEffect(() => {
+    const initialSchedules: Date[][] = [];
+
+    for (let i = 0; i < startDates.length; i += 1) {
+      initialSchedules.push([]);
+    }
+
+    setUserSchedules(initialSchedules);
+  }, [startDates.length]);
+
+  const updateUserSchedules = (scheduleIdx: number) => (schedule: Date[]) => {
```

Comment:
should rename - function doesn't actually update anything, it instead returns a new function